pyne/mesh.py

```diff
@@ -726,7 +727,7 @@ def __init__(self, mesh=None, structured=False,
                 # check for the structured box tag on the instance
                 try:
                     box_tag = self.mesh.tag_get_handle(_BOX_DIMS_TAG_NAME)
-                except types.MB_TAG_NOT_FOUND as e:
+                except Exception as e:
```

Comment:
It's really better to capture specific exceptions to make sure we aren't overlooking an error that should be handled differently.

Reply:
I agree to use specific exceptions. But the problem is `types.MB_TAG_NOT_FOUND` exception is not recognized by python3. Reason not known yet.

Reply:
Yeah I'll look into that and see if we can do something better here...

Reply:
I think this exception can be captured as a `RuntimeError`.

```python
In [1]: from pymoab import core, types                                                                                                            

In [2]: mb = core.Core()                                                                                                                          

In [3]: mb.tag_get_handle(types.CATEGORY_TAG_NAME)                                                                                                
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-3-6275d766220d> in <module>
----> 1 mb.tag_get_handle(types.CATEGORY_TAG_NAME)

pymoab/core.pyx in pymoab.core.Core.tag_get_handle()

pymoab/types.pyx in pymoab.types.check_error()

RuntimeError: MOAB ErrorCode: MB_TAG_NOT_FOUND
```

Here, the category tag doesn't exist because this is a newly created PyMOAB instance and the category tag isn't a default tag.

Reply:
Applied in 0df5d71.